jerry-core/ecma/operations/ecma-typedarray-object.c

```diff
@@ -0,0 +1,941 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ecma-typedarray-object.h"
+#include "ecma-arraybuffer-object.h"
+#include "ecma-function-object.h"
+#include "ecma-builtin-helpers.h"
+#include "ecma-try-catch-macro.h"
+#include "ecma-objects.h"
+#include "ecma-builtins.h"
+#include "ecma-exceptions.h"
+#include "ecma-gc.h"
+#include "ecma-globals.h"
+#include "ecma-helpers.h"
+
+#ifndef CONFIG_DISABLE_TYPEDARRAY_BUILTIN
+
+/** \addtogroup ecma ECMA
+ * @{
+ *
+ * \addtogroup ecmatypedarrayobject ECMA %TypedArray% object related routines
+ * @{
+ */
+
+/**
+ * Create a TypedArray object by given array_length
+ *
+ * See also: ES2015 22.2.1.2.1
+ *
+ * @return pointer to the new typedarray object
+ */
+static ecma_object_t *
+ecma_typedarray_create_object_with_length (ecma_length_t array_length, /**< length of the typedarray */
+                                           ecma_object_t *proto_p, /**< prototype object */
+                                           uint8_t element_size_shift, /**< the size shift of the element length */
+                                           lit_magic_string_id_t class_id) /**< class name of the typedarray */
+{
+  ecma_length_t byte_length = array_length << element_size_shift;
+  ecma_object_t *object_p = ecma_create_object (proto_p,
+                                                sizeof (ecma_extended_object_t),
+                                                ECMA_OBJECT_TYPE_PSEUDO_ARRAY);
+
+  ecma_extended_object_t *ext_object_p = (ecma_extended_object_t *) object_p;
+  ext_object_p->u.pseudo_array.u1.class_id = class_id;
+  ext_object_p->u.pseudo_array.type = ECMA_PSEUDO_ARRAY_TYPEDARRAY;
+  ext_object_p->u.pseudo_array.extra_info = element_size_shift;
+  ext_object_p->u.pseudo_array.u2.arraybuffer = ecma_make_object_value (ecma_arraybuffer_new_object (byte_length));
+  ecma_free_value (ext_object_p->u.pseudo_array.u2.arraybuffer);
+
+  return object_p;
```

Comment:
Perhaps this function could call ecma_typedarray_create_object_with_buffer with the newly created buffer. Or is this small enough function?